src/functionality.py

```diff
@@ -49,24 +150,652 @@ def _validate_command(ctx: commands.Context) -> bool:
+
+            rp_list = rp_id_check()
+            found = []
+
+            for rp in rp_list:
+                if str(1) == rp["ended"]:
+                    found.append(rp)
+
+            return found
+
+        def rp_profile_check_all(val):
+            if val != 3:
+                return
+
+            rp_list = rp_id_check()
+            found = []
+
+            for rp in rp_list:
+                found.append(rp)
+
+            return found
+
+        def timezone_time_check(hour_change: int, inc_time: int):
+            hour_change = hour_change + inc_time
+
+            if hour_change > 23:
+                hour_change = hour_change - 24
+                if len(str(hour_change)) < 2:
+                    hour_change = f"0{hour_change}"
+
+            elif hour_change < 0:
+                hour_change = hour_change + 24
+                if len(str(hour_change)) < 2:
+                    hour_change = f"0{hour_change}"
+
+            if len(str(hour_change)) < 2:
+                hour_change = f"0{hour_change}"
+
+            return hour_change
+
+        # -- Commands Area -- #
+
+        @bot.command(
+            name='ban_id',
+            brief='Bans a Discord User from using the Bot.',
+            help=('Bans a Discord User from using the Bot. There are multiple arguments needed to be filled.'
+                  '\nArguments: $ban_id <discord_id> <ban_length : seconds> <reason> '
+                  '\nExample: $ban_id 332456386946531328 259200 "There must be a open and close quotes for reason."'),
+            hidden=True,
+        )
+        async def ban_id(ctx: commands.Context, user_id: int, ban_length: int = 259200, reason: str = "Unstated Reason"):
+            if not _validate_command(ctx):
+                return
+
+            if not admin_check(ctx):
+                return await ctx.send("`Insufficient Privileges.`")
+
+            ban_ids = ban_id_check()
+
+            try:
+                target = await bot.fetch_user(user_id)
+                print(target.name)
+            except discord.NotFound as p:
+                return await ctx.send(f"`{p}`\n**Please input a valid Discord ID that is in the server.**")
+
+            for dict_ban in ban_ids[1:]:
+                bool_ = int(dict_ban["ended"])
+                bool_ = bool(bool_)
+
+                if str(user_id) == str(dict_ban["discord_id"]) and bool_ is False:
+                    return await ctx.send(f"**{target.name} is already in the list and his ban has not ended.**")
+
+            with open("ban_ids.csv", "a") as file_:
+                fieldnames = ["discord_id", "discord_name", "ban_timestamp", "ban_length", "reason", "ended"]
+                writer = csv.DictWriter(file_, fieldnames=fieldnames)
+                writer.writerow({"discord_id": user_id, "discord_name": target.name, "ban_timestamp": round(time.time()), "ban_length": ban_length, "reason": reason, "ended": 0})
+
+            days = second_to_day(ban_length)
```

Comment:
As is right now, this function will show "0 days" if `ban_length < 86400/2`, which may be a bit confusing to the person that ran the command. Maybe change the wording if the message would display a ban length of 0 days?